config/rbac/role.yaml

```diff
@@ -94,8 +94,10 @@ rules:
   resources:
   - secrets
   verbs:
+  - create
```

Comment:
The controller needs to be able to create `Secrets` to hold the Consul CA cert for each gateway `Deployment`. I used a `Secret` to be consistent w/ consul-k8s which uses a `Secret` for the same value.